src/ucp/core/ucp_worker.c

```diff
@@ -2453,44 +2454,83 @@ static int ucp_worker_keepalive_is_enabled(ucp_worker_h worker)
            (worker->context->config.ext.keepalive_num_eps != 0);
 }
 
+static UCS_F_ALWAYS_INLINE ucp_ep_h
+ucp_worker_keepalive_get_ep(ucp_worker_h worker)
+{
+    ucp_ep_ext_gen_t *ep_ext;
+
+    ep_ext = ucs_container_of(worker->keepalive.iter, ucp_ep_ext_gen_t, ep_list);
+    return ucp_ep_from_ext_gen(ep_ext);
+}
+
+static UCS_F_ALWAYS_INLINE void
+ucp_worker_keepalive_next_ep(ucp_worker_h worker)
+{
+    ucp_ep_h ep;
+
+    worker->keepalive.iter = worker->keepalive.iter->next;
+    if (worker->keepalive.iter == &worker->all_eps) {
+        /* if next list item points to all_eps then step one more time */
+        worker->keepalive.iter = worker->keepalive.iter->next;
+    }
+
+    ucs_assert(worker->keepalive.iter != &worker->all_eps);
+    ep                         = ucp_worker_keepalive_get_ep(worker);
+    worker->keepalive.lane_map = ucp_ep_keepalive_is_enabled(ep) ?
+                                 ucp_ep_config(ep)->key.ep_check_map : 0;
```

Comment:
why not just take ucp_ep_config(ep)->key.ep_check_map ?
why need to use ucp_ep_keepalive_is_enabled both in here and also in ucp_ep_do_keepalive? one of them will be enough